### PR TITLE
style: Just ensures the Register & Login pages match

### PR DESCRIPTION
**Pull request by rschristian**
This was a bit odd as they're pages that are largely duplicates but were written very differently.

### CODE REVIEW

**Comment by rschristian** [comment]
Disabling the tests on the CI for now as they keep breaking and they really do need to be holistically rewritten. For example, the register and login tests are entirely different for some reason, even though they should be near identical forms. 

Fixing the tests every time I fix change the source really isn't to anyone's benefit at the moment. Better to fix later.

**Comment by mutoe** [comment]
I don't think rewrite code in the master branch is the best practice. 

In fact, we should not break the ONLINE page to do any refactor, special we turned off the tests. This will not guarantee that we will have the correct functionality in the future.

Be careful, when you change any existing code and there is no test to cover.

It is best to do large-scale refactoring in other branches, and we need to ensure that our production is normal. There are these benefits:

1. During refactoring, the Preact fans want to preview the page, then they will see an unsound website and may lose confidence.
2. When I want to give up refactoring (for various reasons), I just need to delete the branch instead of rolling back to a commit I don't remember.
3. If I get sick halfway through,  then no one has the context to do the next thing. The ONLINE page will unavailable always.

**Comment by rschristian** [comment]
> I don't think rewrite code in the master branch is the best practice.

The project needed so much work that doing it all in a single PR was unreasonable. You'd have thousands of lines of actual changes with little documentation. Needed to be split out.

> In fact, we should not break the ONLINE page to do any refactor, special we turned off the tests.

The online page is already super broken. Lots of bugs, missing features, and deviations from the spec entirely. I really don't think there's any value in hosting that at this moment. The repo needs a lot of work.

The existing test suite isn't helping, to say the least. I'd say most of the tests don't actually test anything and are only getting in the way of the correct refactoring.

**Comment by rschristian** [comment]
For as long as I need to rework components as much as I did here, due to the massive variation between them, a test suite doesn't offer any utility. I'd spend more time maintaining the old/bad tests than actually making corrections. And then the fixed tests would become super broken in the next hour anyhow. 

We can go through after and fix it up. Save us a lot of time.

File: src/pages/Login.tsx [hunk 1]
```diff
@@ -43,30 +43,29 @@ export default function Register() {
...
 									placeholder="Password"
+									required
+									value={form.password}
 									onInput={e => setForm(prev => ({ ...prev, password: e.currentTarget.value }))}
 								/>
 							</fieldset>
 							<button
 								class="btn btn-lg btn-primary pull-xs-right"
 								disabled={!form.email || !form.password}
-								type="submit"
```
Comment (mutoe) [inline src/pages/Login.tsx:69]: I want to verify whether I can tap <kbd>Enter</kbd> in password input filed will submit the form. (I don't remember the submit button can do this or not), but I found that the ONLINE page was broken. 😂

Comment (rschristian) [reply src/pages/Login.tsx:69]: Submit is the default behavior in a form. https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type

**Comment by mutoe** [comment]
> The existing test suite isn't helping, to say the least. I'd say most of the tests don't actually test anything and are only getting in the way of the correct refactoring.

Maybe yes, this project was when I was new to React/Preact, and I didn't have any front-end testing experience at the time. 

What testing unit test framework do you plan to use in the future? Testing library?

File: src/pages/Login.tsx (hunk 1 above)
Comment (rschristian) [reply src/pages/Login.tsx:69]: `type="submit"` is default within a form, so this does nothing: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-type

Pressing `Enter` works just as you'd expect.

**Comment by rschristian** [comment]
Oh I have absolutely no testing experience myself. I'm not sure if we need another tool, or if we just rework the tests. I've certainly heard good things about Testing Lib, but have no experience.

It's just that a fair few of the tests don't provide much utility IMO. Like they test if a function was called, but in most cases, things have to be very broken for that call to not happen. Like I'd rather check function response and the like. 

Just think we should do a run through at the end and then evaluate what we want to test and what provides solid value.

**Pull request merged**